### ISSUE TITLE
DeepONet Triple Data Shape | if branch & trunk inputs are just time a Parametric Space

### ISSUE THREAD

**Issue by cfd-ai** [lululxvi/deeponet#17]
I have a similar question to https://github.com/lululxvi/deeponet/issues/9.
Imagine you have a transient simulation of a 3D domain and you are interested in an averaged quantity, e.g., Maximum Velocity  V(t) .  Then you would repeat this for various parameters (a parametric space), like Inlet Pressure (many Pi) and Inlet Temperature (many Ti).  To translate this to the DeepONet terminology, I guess:

G(u)(y) =  V(t)(Pi,Ti)
That is to say, 
u  is  t  ::  Input to Branch is time
(y) is (Pi, Ti)  ::  Input to Trunk is pair of (Pi, Ti)

Provided that the t array is always the same for all training, test, and future prediction values, I mean you are always interested in certain time values like [t1=0.1, t2=0.2, t3=0.3, ...], i.e., the objective is to predict V(t) always at the same times for any given (Pi,Ti), 
then I wonder which of the following forms is correct for the DeepONet Triple Data:

```python
X_train=
[
	[
		[ V(t1)(P1,T1), V(t2)(P1,T1), V(t3)(P1,T1)... ] ,
		[ V(t1)(P2,T2), V(t2)(P2,T2), V(t3)(P2,T2)... ] ,
		...
	],
	[
		[ t1, t2, t3... ] ,
		[ t1, t2, t3... ] ,
		...
	]
]

y_train=
[
	[P1 , T1] ,
	[P2 , T2] ,
	...
]
```

OR

```python
X_train=
[
	[
		[ V(t1)(P1,T1) ] ,
		[ V(t2)(P1,T1) ] ,
		[ V(t3)(P1,T1) ] ,
		...
		[ V(t1)(P2,T2) ] ,
		[ V(t2)(P2,T2) ] ,
		[ V(t3)(P2,T2) ] ,
		...
	],
	[
		[t1] ,
		[t2] ,
		[t3] ,
		...
		[t1] ,
		[t2] ,
		[t3] ,
		...
	]
]

y_train=
[
	[P1 , T1] ,
	[P1 , T1] ,
	[P1 , T1] ,
	...
	[P2 , T2] ,
	[P2 , T2] ,
	[P2 , T2] ,
	...
]
```

Basically, train & predict over the entire [t] array together?  or train & predict per each t value?
or both of them are totally wrong?

Summary:
- Time: [t1, t2, t3,..., tn]
- Parameters:  [ [P1,T1] , [T2, P2] , ... , [Tm,Pm] ]

I guess in some form:
- branch:  Time would be the input to the branch net.
- trunk: Parameters would be the input to the branch net.

Am I correct?

Thanks!

_Originally posted by @cfd-ai in https://github.com/lululxvi/deeponet/issues/9#issuecomment-1030532132_

**Comment by lululxvi** [comment]
I am a little confused about the setup. Let us consider a simple: G(u)(t) = s(t), i.e., for any u(t), I want to predict the corresponding s(t). Then t is the input to trunk net, and u(t) is the input to branch net.

**Comment by cfd-ai** [comment]
Thanks @lululxvi.
Then in G(u)(t) = s(t)  form,  for our problem:
- u(t) is the time array, e.g.  [time1, time2, time3, time4, time5]
- t  is a parameter or a group of parameters, e.g.,  [ (Inlet_Pressure, Inlet_Temperature) ] 
- s(t) is Maximum Velocity of the same size of u(t), e.g.  [V_max1, V_max2, V_max3, V_max4, V_max5]

So, if I understand correctly, we won't be able to feed this because in DeepONet,  t (input to trunk net) has to be of the same size of u(t) (input to the branch net), am I correct?  While here the input to the branch net has 5 elements but the input to the trunk net is only 1. 

Am I missing something?

**Comment by lululxvi** [comment]
No. it works. Branch and trunk are two independent networks.
